src/Microsoft.DotNet.CodeFormatting/Options/StringSet.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    [Serializable]
+    public class StringSet : HashSet<string>
```

Comment:
HashSet<T> instantiations are not serializable. We have to define this type in order to support binary serialization.